modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradePortletDisplayCheck.java

```diff
@@ -0,0 +1,143 @@
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradePortletDisplayCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
+
+		return addNewImports(fileName, newContent);
+	}
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		if (fileName.endsWith(".jsp")) {
+			Matcher getPortletInstanceConfigurationMatcher =
+				_getPortletInstanceConfigurationPattern.matcher(content);
+
+			while (getPortletInstanceConfigurationMatcher.find()) {
+				return _newContent(
+					newContent, fileName,
+					getPortletInstanceConfigurationMatcher, newContent,
+					"themeDisplay");
+			}
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			Matcher getPortletInstanceConfigurationMatcher =
+				_getPortletInstanceConfigurationPattern.matcher(
+					javaMethodContent);
+
+			while (getPortletInstanceConfigurationMatcher.find()) {
+				Matcher themeDisplayMatcher = _themeDisplayPattern.matcher(
+					javaMethodContent);
+
+				if (themeDisplayMatcher.find()) {
+					newContent = _newContent(
+						javaMethodContent, fileName,
+						getPortletInstanceConfigurationMatcher, newContent,
+						themeDisplayMatcher.group(1));
+				}
+				else {
+					newContent = _newContent(
+						javaMethodContent, fileName,
+						getPortletInstanceConfigurationMatcher, newContent,
+						getPortletInstanceConfigurationMatcher.group(1) +
+							".getThemeDisplay()");
+				}
+			}
+		}
+
+		return newContent;
+	}
+
+	@Override
+	protected String[] getNewImports() {
+		return new String[] {
+			"com.liferay.portal.configuration.module.configuration." +
+				"ConfigurationProviderUtil"
+		};
+	}
+
+	@Override
+	protected String[] getValidExtensions() {
+		return new String[] {"java", "jsp"};
+	}
+
+	private String _newContent(
+		String content, String fileName, Matcher matcher, String newContent,
+		String newParameters) {
+
+		String methodCall = JavaSourceUtil.getMethodCall(
+			content, matcher.start());
+
+		String variableName = getVariableName(methodCall);
+
+		if (!hasClassOrVariableName(
+				"PortletDisplay", newContent, fileName, methodCall) &&
+			!variableName.contains("portletDisplay")) {
+
+			return newContent;
+		}
+
+		String indent = JavaSourceUtil.getIndent(methodCall);
```

Comment:
@TamyrisTorres, please declare this variable after the `if` below

Reply:
resolved